pkg/trafficpolicy/trafficpolicy.go

```diff
@@ -131,14 +133,28 @@ func MergeInboundPolicies(allowPartialHostnamesMatch bool, original []*InboundTr
 }
 
 // MergeOutboundPolicies merges two slices of *OutboundTrafficPolicies so that there is only one traffic policy for a given set of a hostnames
-func MergeOutboundPolicies(original []*OutboundTrafficPolicy, latest ...*OutboundTrafficPolicy) []*OutboundTrafficPolicy {
+// allowPartialHostnamesMatch when set to true merges outbound policies by partially comparing (subset of one another) the hostnames of the original traffic policy to the latest traffic policy
+// Two outbound traffic policies can be merged by comparing their hostnames either partially or completely. A partial match on hostnames should be allowed for the following scenarios :
+// when a policy having its hostnames from a host header needs to be merged with other outbound policies
+// This is because there will be a single hostname (from the host header) and there is a possibility that this hostname is part of an existing traffic policy
+// hence the rules need to be merged
+func MergeOutboundPolicies(allowPartialHostnamesMatch bool, original []*OutboundTrafficPolicy, latest ...*OutboundTrafficPolicy) []*OutboundTrafficPolicy {
 	for _, l := range latest {
 		foundHostnames := false
 		for _, or := range original {
-			if reflect.DeepEqual(or.Hostnames, l.Hostnames) {
-				foundHostnames = true
-				mergedRoutes := mergeRoutesWeightedClusters(or.Routes, l.Routes)
-				or.Routes = mergedRoutes
+			if !allowPartialHostnamesMatch {
+				if reflect.DeepEqual(or.Hostnames, l.Hostnames) {
+					foundHostnames = true
+					mergedRoutes := mergeRoutesWeightedClusters(or.Routes, l.Routes)
+					or.Routes = mergedRoutes
+				}
+			} else {
+				if hostsUnion := slicesUnion(or.Hostnames, l.Hostnames); len(hostsUnion) > 0 {
```

Comment:
It would be so good to put a comment right above `sliceUnion` to explain what it does.
Just reading the `sliceUnion` - I'd me inclined to thing that we just have the union of or.Hostnames and l.Hostnames -- but it is a bit more complicated than that...

Reply:
+1